omod-common/src/main/java/org/openmrs/module/webservices/rest/web/api/RestService.java

```diff
@@ -27,62 +27,77 @@
 	/**
 	 * Parses a representation requested by the client via the http request
 	 * 
-	 * @param requested
+	 * @param requested representation
 	 * @return
 	 * <strong>Should</strong> get ref representation when specified
 	 * <strong>Should</strong> get default representation when specified
 	 * <strong>Should</strong> get full representation when specified
 	 * <strong>Should</strong> get a named representation when specified
 	 */
-	public Representation getRepresentation(String requested);
+	Representation getRepresentation(String requested);
 	
+	/**
+	 * Gets resource by name.
+	 *
+	 * @param name the name
+	 * @return the resource by name
+	 * @throws APIException the api exception
+	 */
 	Resource getResourceByName(String name) throws APIException;
 	
 	/**
-	 * Auto generated method comment
+	 * Gets resource by name or uuid.
 	 * 
-	 * @param supportedClass
-	 * @return
-	 * @throws APIException
+	 * @param str the name or uuid
+	 * @return the resource by name or uuid
+	 * @throws APIException the api exception
+	 */
+	Resource getResourceByNameOrUuid(String str) throws APIException;
+	
+	/**
+	 * Gets resource by supported class
+	 * 
+	 * @param supportedClass the supported class
+	 * @return resource by supported class
+	 * @throws APIException the api exception
 	 */
 	Resource getResourceBySupportedClass(Class<?> supportedClass) throws APIException;
 	
 	/**
 	 * Returns a search handler, which supports the given resource and the map of parameters and
 	 * values.
-	 * 
-	 * @param resourceName
-	 * @param parameters
+     * @param resourceName the resource name
+     * @param parameters   the parameters
 	 * @return searchHandler or <code>null</code> if no match
-	 * @throws APIException
+	 * @throws APIException the api exception
 	 */
 	SearchHandler getSearchHandler(String resourceName, Map<String, String[]> parameters) throws APIException;
 	
 	/**
 	 * Returns all search handlers supporting a resource
 	 * 
-	 * @param resourceName
-	 * @return
+	 * @param resourceName the resource name
+	 * @return search handlers
 	 */
 	Set<SearchHandler> getSearchHandlers(String resourceName);
 	
 	/**
 	 * Returns all {@link DelegatingResourceHandler}s
 	 * 
 	 * @return list of {@link DelegatingResourceHandler}s
-	 * @throws APIException
+	 * @throws APIException the api exception
 	 */
-	public List<DelegatingResourceHandler<?>> getResourceHandlers() throws APIException;
+	List<DelegatingResourceHandler<?>> getResourceHandlers() throws APIException;
 	
 	/**
 	 * Initializes all Resources and Search handlers for use; called after module startup
 	 */
-	public void initialize();
+	void initialize();
```

Comment:
i don't think we need to change the visibility of the method

Reply:
The modifier was [redundant ](https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.html#:~:text=Rationale%3A%20The%20Java%20Language%20Specification%20strongly%20discourages%20the%20usage%20of%20public%20and%20abstract%20for%20method%20declarations%20in%20interface%20definitions%20as%20a%20matter%20of%20style.) on that method.

Reply:
Shouldn't these general improvements have a ticket and pull request of their own?

Reply:
Reverting changes.